swarm/network/stream/common_test.go

```diff
@@ -317,3 +320,30 @@ func createTestLocalStorageForID(id enode.ID, addr *network.BzzAddr) (storage.Ch
 	}
 	return store, datadir, nil
 }
+
+// watchDisconnections receives simulation peer events in a new goroutine and sets atomic value
+// disconnected to true in case of a disconnect event.
+func watchDisconnections(ctx context.Context, sim *simulation.Simulation) (disconnected atomic.Value) {
```

Comment:
I tried to understand this function better. I noticed the following  in `atomic.Value`'s doc: `Once Store has been called, a Value must not be copied.`. The function's return parameter copies `Value`. The sub-goroutine might call `disconnected.Store()` before. Do you think that's true?

Reply:
Yes, I did not know about this restriction on atomic.Value. Because of thinks like this I avoid atomic function, but here I decide to use it. Do you think that returning a pointer to atomic.Value would solve it?

Reply:
Good question. I think so.

Maybe you could just use a `type` with fields and methods. Then you could avoid the copy and also you could DRY out/hide `yes, ok := disconnected.Load().(bool); ok && yes ` checks into a method. What do you think?

Reply:
Right, avoiding type check is good too. I've pushed an alternative solution.